server/app/services/languageProcessor/LUIS.java

```diff
@@ -0,0 +1,102 @@
+package services.languageProcessor;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import play.Configuration;
+import play.api.Play;
+import play.libs.Json;
+import play.libs.ws.WSClient;
+import play.libs.ws.WSRequest;
+import play.libs.ws.WSResponse;
+import play.mvc.Result;
+import services.IntentEntity;
+import services.queryHandler.JiraApiFetchInfo;
+import java.util.concurrent.CompletionStage;
+
+import static play.mvc.Results.ok;
+
+/**
+ * Created by sabinapokhrel on 11/21/16.
+ */
+public class LUIS {
+
+  private Configuration configuration = Play.current().injector().instanceOf(Configuration.class);
+
+  private String query;
+  private WSClient ws;
+
+
+  public LUIS(String query, WSClient ws) {
+    this.query = query;
+    this.ws = ws;
+  }
+
+  public JsonNode handleQuery(){
+
+    return asyncGET(query);
+  }
+
+  /**
+   * This method connects to the LUIS API and sends a query.
+   * @param query is the question by a user to POET on slack channel.
+   * @return the result of classification performed by LUSI on the asked query as a JSON object.
+   */
+  public JsonNode asyncGET(String query) {
+
+    CompletionStage<JsonNode> responsePromise = getTask(query);
+    try{
+      JsonNode node = responsePromise.toCompletableFuture().get();
+      return taskMapping(node);
+    }catch(Exception e){
+      System.out.println("Exception: " + e.toString());
+      return null;
+    }
+  }
+
+  /**
+   * Connects to LUIS API using HTTPS request.
+   * LUIS API processes the query and returns intent and entities as JSON object
+   * @param query
+   * @return
+   */
+  public CompletionStage<JsonNode> getTask(String query) {
```

Comment:
Consider whether method and the one below it should be `private`. Do you expect users of this class to call these two methods directly? Or should they be calling `ayncGet`?